From: coolsnowwolf <coolsnowwolf@gmail.com>
Subject: [PATCH] build: detect OpenSSL before selecting version 3 API

LibreSSL's version output also has a 3.x version in its second field. Check
the implementation name before enabling the OpenSSL 3 API, otherwise host
tools mix incompatible initialization and key generation code paths.

Signed-off-by: coolsnowwolf <coolsnowwolf@gmail.com>
---
 make_helpers/build_macros.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/make_helpers/build_macros.mk
+++ b/make_helpers/build_macros.mk
@@ -272,7 +272,7 @@
     $(eval OPENSSL_CURRENT_VER = $(word 2, ${OPENSSL_INFO}))
     $(eval OPENSSL_CURRENT_VER_MAJOR = $(firstword $(subst ., ,$(OPENSSL_CURRENT_VER))))
     # If OpenSSL version is 3.x, then set USING_OPENSSL3 flag to 1
-    $(if $(filter 3,$(OPENSSL_CURRENT_VER_MAJOR)), $(eval USING_OPENSSL3 = 1))
+    $(if $(and $(filter OpenSSL,$(word 1,$(OPENSSL_INFO))),$(filter 3,$(OPENSSL_CURRENT_VER_MAJOR))), $(eval USING_OPENSSL3 = 1))
 endef
 
 ################################################################################
